crates/noirc_abi/src/lib.rs

```diff
@@ -155,12 +155,24 @@ impl Abi {
     pub fn encode(
         self,
         inputs: &BTreeMap<String, InputValue>,
-        allow_undefined_return: bool,
+        skip_output: bool,
     ) -> Result<Vec<FieldElement>, AbiError> {
-        let param_names = self.parameter_names();
+        // Condition that specifies whether we should filter the "return"
+        // parameter. We do this in the case that it is not in the `inputs`
+        // map specified.
+        //
+        // See Issue #645 : Adding a `public outputs` field into acir and
+        // the ABI will clean up this logic
+        // For prosperity; the prover does not know about a `return` value
+        // so we skip this when encoding the ABI
+        let return_condition =
```

Comment:
This was also brought up in the previous issue. The issue is fixed by changing the behaviour, else we will need to employ machinery to optimise `constrain return == x` with the binary solver.

There is the option to not constrain the return value, but this is not intuitive and is also different to what it was doing before.

Reply:
You do not need to add a constraint, so you do not need the binary solver. You just need to check that the computed return value matches the return value in prover.toml.

Reply:
Yep, I mentioned this in the last step. I don't think this should be the accepted behaviour.

Reply:
In that case could you create an issue rather than changing the behaviour?

Reply:
I'm not sure I get your point, the behavior will be changed with your suggestion also. The other fix is to introduce a Binary solver to solve for the case when you have multiple unknowns.

Reply:
There is already an issue opened for this [here](https://github.com/noir-lang/noir/issues/624) . There it is discussed that the behavior you are suggesting can fit under testing

Reply:
As discussed I believe there is a real use case, which can be done without this feature by adding parameters and constraints, making the result less performant and more cumbersome (instead of using the return="", you have to write a custom parameter and constrain it properly).
But since you still disagree and we need to move, I am closing this conversation.